pkg/store/store_test.go

```diff
@@ -0,0 +1,196 @@
+	assert.True(t, exists)
+	assert.Equal(t, testContent, string(content))
+
+	// Test adding non-existent file
+	err = store.AddFile("/path/that/does/not/exist")
+	assert.Error(t, err)
+	assert.Contains(t, err.Error(), "file does not exist")
+}
+
+func TestStore_RemoveFile(t *testing.T) {
+	tempDir := t.TempDir()
+	testFile := filepath.Join(tempDir, "test.txt")
+
+	err := os.WriteFile(testFile, []byte("content"), 0o644)
+	require.NoError(t, err)
+
+	ctx := context.Background()
+	store := New(logr.Discard(), ctx, 0) // 0 interval disables periodic refresh
+
+	// Add and then remove file
+	err = store.AddFile(testFile)
+	require.NoError(t, err)
+
+	store.RemoveFile(testFile)
+
+	// Verify file was removed
+	_, exists := store.GetContent(testFile)
+	assert.False(t, exists)
+}
+
+func TestStore_GetContent(t *testing.T) {
+	tempDir := t.TempDir()
+	testFile := filepath.Join(tempDir, "test.txt")
+	testContent := "test content"
+
+	err := os.WriteFile(testFile, []byte(testContent), 0o644)
+	require.NoError(t, err)
+
+	ctx := context.Background()
+	store := New(logr.Discard(), ctx, 0) // 0 interval disables periodic refresh
+	err = store.AddFile(testFile)
+	require.NoError(t, err)
+
+	// Test GetContent
+	content, exists := store.GetContent(testFile)
+	require.True(t, exists)
+	assert.Equal(t, testContent, string(content))
+
+	// Test non-existent file
+	_, exists = store.GetContent("/nonexistent/path")
+	assert.False(t, exists)
+}
+
+func TestStore_PeriodicRefresh(t *testing.T) {
+	tempDir := t.TempDir()
+	testFile := filepath.Join(tempDir, "test.txt")
+	originalContent := "original"
+	updatedContent := "updated"
+
+	// Create test file
+	err := os.WriteFile(testFile, []byte(originalContent), 0o644)
+	require.NoError(t, err)
+
+	// Start periodic refresh with short interval
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	store := New(logr.Discard(), ctx, 50*time.Millisecond)
+	err = store.AddFile(testFile)
+	require.NoError(t, err)
+
+	// Ensure file is in store
+	content, exists := store.GetContent(testFile)
+	require.True(t, exists)
+	assert.Equal(t, originalContent, string(content))
+
+	// Wait a bit, then update file
+	time.Sleep(10 * time.Millisecond)
+	err = os.WriteFile(testFile, []byte(updatedContent), 0o644)
+	require.NoError(t, err)
+
+	// Wait for periodic refresh to pick up the change
+	assert.Eventually(t, func() bool {
+		content, exists := store.GetContent(testFile)
+		return exists && string(content) == updatedContent
+	}, 200*time.Millisecond, 10*time.Millisecond)
+}
+
+func TestStore_ConcurrentAccess(t *testing.T) {
+	tempDir := t.TempDir()
+	testFile := filepath.Join(tempDir, "test.txt")
+
+	err := os.WriteFile(testFile, []byte("initial"), 0o644)
+	require.NoError(t, err)
+
+	ctx := context.Background()
+	store := New(logr.Discard(), ctx, 0) // 0 interval disables periodic refresh
+	err = store.AddFile(testFile)
+	require.NoError(t, err)
+
+	// Test concurrent read/write access
```

Comment:
how are we testing write logic here? wouldn't we want to have a "write" event after which we expect both readers are consistent?

Reply:
updated to use fssync